nilearn/experimental/plotting/_surface_plotting.py

```diff
@@ -0,0 +1,268 @@
+"""Experimental Surface plotting."""
+
+from __future__ import annotations
+
+import numpy
+
+from nilearn import plotting as old_plotting
+from nilearn._utils import fill_doc
+from nilearn.experimental.surface import Mesh, PolyMesh, SurfaceImage
+
+
+@fill_doc
+def plot_surf(
+    surf_map: SurfaceImage | str | numpy.array | None = None,
+    hemi: str = "left",
+    surf_mesh: (
+        str | list[numpy.array, numpy.array] | Mesh | PolyMesh | None
+    ) = None,
+    bg_map: str | numpy.array | SurfaceImage | None = None,
+    **kwargs,
+):
+    """Plot surfaces with optional background and data.
+
+    Parameters
+    ----------
+    surf_map : SurfaceImage or :obj:`str` or numpy.ndarray, optional
+
+    surf_mesh : :obj:`str` or :obj:`list` of two numpy.ndarray \
+                or Mesh or PolyMesh, optional
+
+    %(hemi)s
+
+    bg_map : str or numpy.ndarray or SurfaceImage, optional
+    """
+    bg_map = _check_bg_map(bg_map, hemi)
+
+    if not isinstance(surf_map, SurfaceImage):
+        return old_plotting.plot_surf(
+            surf_mesh=surf_mesh,
+            surf_map=surf_map,
+            hemi=hemi,
+            bg_map=bg_map**kwargs,
+        )
+
+    if surf_mesh is None:
+        surf_mesh = surf_map.mesh
+
+    surf_map._check_hemi(hemi)
+    _check_hemi_present(surf_mesh, hemi)
```

Comment:
this will fail if surf_mesh is not a PolyMesh
fixing this will probably need better define the PolyMesh class

Reply:
Can you clarify the discussion here ?

Reply:
`_check_hemi_present` is just to make sure that the "PolyMesh" dict has the correct hemisphere

But most of our new code at the moment, does not use the PolyMesh class but simple dictionaries instead, so we cannot do a `isinstance(surf_mesh, PolyMesh)`. So we should probably develop the PolyMesh class, because at the moment it is only `PolyMesh = Dict[str, Mesh]`.

Also at this stage `surf_mesh` can also be a str, list of two numpy.ndarray, Mesh and none of those cases are handled.

Reply:
Not I could start making those changes in this PR but I was getting afraid of having an even bigger PR